File: examples/docs-code-snippets/src/main/java/examples/java/recipes/RecipeFireEventAndResolve.java
package examples.java.recipes;

import com.ing.baker.recipe.javadsl.InteractionFailureStrategy;
import com.ing.baker.recipe.javadsl.Recipe;

public class RecipeFireEventAndResolve {

    public final static Recipe recipe = new Recipe("example")
        .withDefaultFailureStrategy(
            InteractionFailureStrategy.FireEventAndResolve("MyEvent")
        );
}

File: examples/docs-code-snippets/src/main/scala/examples/scala/recipes/RecipeFireEventAndResolve.scala
package examples.scala.recipes

import com.ing.baker.recipe.common.InteractionFailureStrategy
import com.ing.baker.recipe.scaladsl.Recipe

object RecipeFireEventAndResolve {
  val recipe: Recipe = Recipe("example")
    .withDefaultFailureStrategy(
      InteractionFailureStrategy.FireEventAndResolve(Some("MyEvent"))
    )
}